jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -324,23 +326,61 @@ def plot_selected_points(self):
                                                            getattr(y, 'value', y))
 
     def vue_zoom_in(self, *args, **kwargs):
-        """This function will zoom into the image based on the selected points"""
+
+        self.zoom_to_selected(return_bounding_box=True)
+
+    def zoom_to_selected(self, padding=0.02, return_bounding_box=False):
+        """
+        Zoom to a region containing the currently selected points in the catalog.
+
+        Parameters
+        ----------
+        padding : float, optional
+            A fractional value representing the padding around the bounding box
+            of the selected points. It is applied as a proportion of the largest
+            dimension of the current extent of loaded data. Defaults to 0.02.
+        return_bounding_box : bool, optional
+            If True, returns the bounding box of the zoomed region as
+            ((x_min, x_max), (y_min, y_max)). Defaults to False.
+
+        Returns
+        -------
+        tuple of tuple or None
+            If there are activley selected rows, and ``return_bounding_box`` is
+            True, returns a tuple containing the bounding
+            box coordinates: ((x_min, x_max), (y_min, y_max)).
+            Otherwise, returns None.
+
+        """
+
+        viewer = self.app._jdaviz_helper._default_viewer
```

Comment:
maybe mention in the docstring that this will only apply to the default viewer?  And maybe we need a follow-up ticket (if one doesn't already exist) for choosing which viewer(s)?